docs/advanced-features/custom-document.md

```diff
@@ -92,10 +92,12 @@ export default MyDocument
 You can use the built-in `DocumentContext` type and change the file name to `./pages/_document.tsx` like so:
 
 ```tsx
-import Document, { DocumentContext } from 'next/document'
+import Document, { DocumentContext, DocumentInitialProps } from 'next/document'
 
 class MyDocument extends Document {
-  static async getInitialProps(ctx: DocumentContext) {
+  static async getInitialProps(
+    ctx: DocumentContext
+  ): Promise<DocumentInitialProps> {
```

Comment:
Afaik TS does a very good job at inferring the type from the `getInitialProps` method that's defined in `Document`, right?

Reply:
hmm, I might have had a out of date TS.